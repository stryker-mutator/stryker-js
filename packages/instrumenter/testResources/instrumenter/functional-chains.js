foo?.bar?.trim?.();

baz?.trim();

qux.trim().substring(3);

quux.trim?.().substring(3);

