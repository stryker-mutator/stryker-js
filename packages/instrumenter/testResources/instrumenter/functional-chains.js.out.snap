// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instrumenter integration should be able to instrument functional chains 1`] = `
"function stryNS_9fa48() {
  var g = new Function(\\"return this\\")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

stryMutAct_9fa48(\\"3\\") ? foo.bar?.trim?.() : stryMutAct_9fa48(\\"2\\") ? foo?.bar.trim?.() : stryMutAct_9fa48(\\"1\\") ? foo?.bar?.trim() : stryMutAct_9fa48(\\"0\\") ? foo?.bar : (stryCov_9fa48(\\"0\\", \\"1\\", \\"2\\", \\"3\\"), foo?.bar?.trim?.());
stryMutAct_9fa48(\\"5\\") ? baz.trim() : stryMutAct_9fa48(\\"4\\") ? baz : (stryCov_9fa48(\\"4\\", \\"5\\"), baz?.trim());
stryMutAct_9fa48(\\"7\\") ? qux.substring(3) : stryMutAct_9fa48(\\"6\\") ? qux.trim() : (stryCov_9fa48(\\"6\\", \\"7\\"), qux.trim().substring(3));
stryMutAct_9fa48(\\"10\\") ? quux.trim().substring(3) : stryMutAct_9fa48(\\"9\\") ? quux.substring(3) : stryMutAct_9fa48(\\"8\\") ? quux.trim?.() : (stryCov_9fa48(\\"8\\", \\"9\\", \\"10\\"), quux.trim?.().substring(3));"
`;
